app/build.gradle

```diff
@@ -17,6 +17,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    useLibrary 'org.apache.http.legacy'
```

Comment:
Well, our external storage (/sdcard/data/Android) is inbetween of these; we can read/write there without any permission and everyone else can read/write there if has storage permission.

Maybe content provider would be another solution, but I'm not sure if they are well supported by pdf readers.